test/syncenginetestutils.cpp

```diff
@@ -288,7 +288,11 @@ FakePropfindReply::FakePropfindReply(FileInfo &remoteRootFileInfo, QNetworkAcces
     auto writeFileResponse = [&](const FileInfo &fileInfo) {
         xml.writeStartElement(davUri, QStringLiteral("response"));
 
-        xml.writeTextElement(davUri, QStringLiteral("href"), prefix + QString::fromUtf8(QUrl::toPercentEncoding(fileInfo.path(), "/")));
+        QString url = prefix + QString::fromUtf8(QUrl::toPercentEncoding(fileInfo.path(), "/"));
+        if (!url.endsWith(QChar('/'))) {
```

Comment:
why is this change really needed ?

Reply:
otherwise, an assert will be triggered during the discovery phase because the URL is not in the expected format. probably it was before no problem because we didn't had subfolders.